Google.Api.Gax.Grpc.Tests/ClientBuilderBaseTest.GetEffectiveEndpointTest.cs

```diff
@@ -0,0 +1,91 @@
+﻿/*
+ * Copyright 2024 Google Inc. All Rights Reserved.
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file or at
+ * https://developers.google.com/open-source/licenses/bsd
+ */
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Google.Api.Gax.Grpc.Tests
+{
+    public partial class ClientBuilderBaseTest
+    {
+        public class GetEffectiveEndpointTest
+        {
+            private const string DefaultUniverseDomain = "googleapis.com";
+            private const string CustomUniverseDomain = "custom.domain.com";
+            private const string DefaultEndpointAtDefaultDomain = "service1.googleapis.com";
+            private const string DefaultEndpointNotAtDefaultDomain = "service1";
+            private const string EffectiveEndpointAtCustomUniverseDomain = "service1.custom.domain.com";
+
+            public static TheoryData<string, string, string> EffectiveEndpointWillHaveValue =>
+                new TheoryData<string, string, string>
+                {
+                    // default endpoint, universe domain, expected effective endpoint
```

Comment:
Nit: default => Default
(I acknowledge this is my OCD around comments more than anything rational...)